pdf/pdfrenderthread.cpp

```diff
@@ -483,7 +483,12 @@ void PDFRenderThreadQueue::processPendingJob()
     case PDFJob::LoadDocumentJob:
         d->loadFailure = false;
         break;
+    case PDFJob::UnLockDocumentJob:
+        job->m_document = d->document;
+        break;
     default:
+        if (d->document->isLocked())
+            return;
```

Comment:
Default case leaks memory? Should warn? Could also note code changes in commit message and why they were done.